core/tests/test_utils.py

```diff
@@ -414,6 +417,104 @@ def wrapper(
     return decorator
 
 
+@contextlib.contextmanager
+def swap_get_platform_parameter_value_function(
+    platform_parameter_name_value_tuples: List[
+        Tuple[
+            platform_parameter_list.ParamName,
+            platform_parameter_domain.PlatformDataTypes
+        ]
+    ]
+) -> Iterator[None]:
+    """Mocks get_platform_parameter_value function within the context of a
+    'with' statement. get_platform_parameter_value will return the value of
+    the platform parameter if the parameter is present in the
+    platform_parameter_names list.
+
+    Args:
+        platform_parameter_name_value_tuples: List[Tuple[ParamName,
+            PlatformDataTypes]]. The list of the names of the platform
+            parameters and their corresponding values that will be enabled.
+
+    Yields:
+        context. The context with function replaced.
+    """
+    def mock_get_platform_parameter_value(
+        parameter_name: str
+    ) -> platform_parameter_domain.PlatformDataTypes:
+        """Mocks get_platform_parameter_value function to return the value of
+        the platform parameter if the parameter is present in the
+        platform_parameter_names list.
+
+        Args:
+            parameter_name: str. The name of the platform parameter whose
+                value is required.
+
+        Returns:
+            PlatformDataTypes. The value of the platform parameter if the
+            parameter is present in the platform_parameter_names list.
+        """
+        platform_parameter_name_value_dict = dict(
```

Comment:
What if a parameter that's retrieved is not specified in the given list of tuples?